src/Symfony/Bundle/SecurityBundle/Resources/config/guard.xml

```diff
@@ -29,6 +29,7 @@
             <argument /> <!-- User Provider -->
             <argument /> <!-- Provider-shared Key -->
             <argument /> <!-- User Checker -->
+            <argument type="service" id="security.password_encoder" />
```

Comment:
i tend to believe `DaoAuthenticationProvider` could also inject the `UserPasswordEncoder`, to save a few lines (currently it uses the factory directly)

Reply:
we already inject the factory since 3.4 here, not worth it